xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/macros.vm

```diff
@@ -2322,6 +2328,31 @@ $string##
 #end
 
 
+#**
+* Pulls all the JavaScript and CSS resources needed by the page picker.
+*#
+#macro (pagePicker_import)
+  #picker_import
+  #set ($discard = $xwiki.jsfx.use('uicomponents/suggest/suggestPages.js',
+    {'forceSkinAction': true, 'language': $xcontext.locale}))
+#end
+
+#macro (pagePicker_input $multiSelect $parameters)
```

Comment:
I wouldn't keep ``$multiSelect`` as separate parameter. I know the ``userPicker`` macro has this but:

* when I added the ``userPicker`` macro 3 years ago I was thinking that ``$parameters`` are just attributes to pass to the picker input.
* with the new user picker it's not just attributes but I kept the "signature" of the macro to not break existing calls.

Thus I would remove the ``pagePicker_input`` macro and simply call

    #suggestInput($parameters)

in ``pagePicker``, which has a single parameter ``$parameters``. WDYT?

Reply:
Yes it makes more sense to have the `multiple` parameter directly in the `$parameters` map.